### PR TITLE
fix:made class that returns string instead of entity we had there

### PR DESCRIPTION
**Pull request by ReneVajda1**
4 changes were implemented here

1 UserServiceResponse class created to have response message as String
2 modified UserService interface to use the 1
3 updated UserServiceImpl to match interface 
4 updated controller to handle UserServiceResponse and convert ResponseEntity<String> as we needed 

again just possible solution check it tell me what u think

### CODE REVIEW

**Comment by ReneVajda1** [comment]
not working .. closing for now .. will try look at it later

File: Backend-API/src/main/java/com/avangers/backendapi/controllers/UserController.java [hunk 1]
```diff
@@ -3,23 +3,28 @@
...
-@CrossOrigin("http://localhost:5173/")
 @RestController
+@CrossOrigin("http://localhost:5173/")
 @RequiredArgsConstructor
 public class UserController {
 
     private final UserService userService;
 
     @PostMapping("/register")
     public ResponseEntity<String> register(@Valid @RequestBody RegisterUserDTO registerUserDTO) {
```
Comment (markkovari) [inline Backend-API/src/main/java/com/avangers/backendapi/controllers/UserController.java:24]: I do not really like returning with anything other than JSON here. Can we create a `ReponseDTO` class/record to make things deserialisable on the frontend and other clients?

File: Backend-API/src/main/java/com/avangers/backendapi/services/UserServiceResponse.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class UserServiceResponse {
+    private String message;
+}
```
Comment (markkovari) [inline Backend-API/src/main/java/com/avangers/backendapi/services/UserServiceResponse.java:12]: Oh my bad, this is what I meant. It should be called something specific. Not just response, but what kind of response it is?! 🤔

File: Backend-API/src/main/java/com/avangers/backendapi/controllers/UserController.java [hunk 3]
```diff
@@ -3,23 +3,28 @@
...
-        return userService.addUser(registerUserDTO);
+        UserServiceResponse response = userService.addUser(registerUserDTO);
```
Comment (markkovari) [inline Backend-API/src/main/java/com/avangers/backendapi/controllers/UserController.java:25]: I would rather throw a custom made error and handle it with a controlleradvice some place else.
https://spring.io/blog/2013/11/01/exception-handling-in-spring-mvc

**Review by markkovari — APPROVED** [review]

**Review by Normelik — APPROVED** [review]

**Pull request merged**